lib/matplotlib/axis.py

```diff
@@ -879,6 +879,9 @@ def clear(self):
         - registered callbacks
         """
         self.label._reset_visual_defaults()
+        self.label.set_color(mpl.rcParams['axes.labelcolor'])
```

Comment:
Shouldn't this all happen in `_reset_visual_defaults`?

Reply:
`self.label` is simply a `Text` instance, so `self.label._reset_visual_defaults()` looks, for example, at the `text.color` rcParams instead of the `axes.labelcolor` rcParams.

`self.label._reset_visual_defaults()` does accept input arguments to override text defaults (although arguably that that doesn't make complete sense given the name of the method).  So, for color in specific, one could call `self.label._reset_visual_defaults(color=mpl.rcParams['axes.labelcolor'])`.  For `axes.labelsize` and `axes.labelweight`, it takes a little more effort because those have to be passed in through the `fontproperties` keyword argument.

Ultimately, I decided that it was overly complicated to pass in those arguments to `_reset_visual_defaults()` or to modify that method for `self.label` to be aware of the defaults for an axis label rather than the defaults for text.  The approach that I took in this PR felt more straightforward.

I have added a clarifying comment for future maintainers.